lib/hyrax/transactions/steps/save_collection_logo.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # Adds logo info via `ChangeSet`.
+      #
+      # During the update collection process this step is called to update the file(s)
+      # to be used as logo(s) for the collection.
+      #
+      class SaveCollectionLogo
```

Comment:
Same for this step.  It needs tests.  You might be able to copy in the tests from the controller and make some minor adjustments for them to work for the transaction code.